task_3/server.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+const bodyParser = require('body-parser');
+const path = require('path');
+const {
+  toJSON,
+  saveEvent,
+  updateEvents,
+  createStream,
+  deleteEvent,
+} = require('./utils');
+const { log } = require('../logger');
+
+const FILE = 'events.csv';
+const filePath = path.join('.', 'files', FILE);
+
+app.use(bodyParser.json());
+
+app.get('/events', async (req, res) => {
+  try {
+    const events = await toJSON(filePath);
+    const reqLocation = req.query.location;
+
+    if (reqLocation) {
+      return res.json(events.filter(({ location }) => location.toLowerCase() === reqLocation.toLowerCase()));
```

Comment:
Potentially we could avoid loading all the events into the memory by filtering needed events while reading data from the file (using streams).